test/org/apache/catalina/valves/TestLoadBalancerDrainingValve.java

```diff
@@ -238,6 +238,8 @@ private void runValve(String jkActivation,
                 expectedCookie.setPath(cookieConfig.getPath());
                 expectedCookie.setMaxAge(0);
 
+                EasyMock.expect(request.isSecure()).andReturn(true);
```

Comment:
Should this be `expect(request.isSecure() || cookieConfig.isSecure())`?

Reply:
Feel free to add it. Makes no difference though, since || is short-circuit.

Reply:
Actually, I think we need a second test method to have coverage for the `cookieConfig.isSecure()` case. E.g. `runValve()` could have an additional boolean/enum parameter that decides whether to prepare `request.isSecure()` or `cookieConfig.isSecure()`.

In addition I don't see any new code that verifies that the `expectedCookie` has `isSecure() == true`